src/main/java/oss/fosslight/config/CustomAuthenticationProvider.java

```diff
@@ -34,102 +36,118 @@
+      loginSuccess = checkByADUser(user_id, user_pw);
+    } else {
+      loginSuccess = checkSystemUser(user_id, user_pw);
+    }
+
+    if (loginSuccess) {
+      List<GrantedAuthority> roles = new ArrayList<GrantedAuthority>();
+      T2Users user = new T2Users();
+      user.setUserId(user_id);
+      T2Users getUser = userService.getUserAndAuthorities(user);
+
+      for (T2Authorities auth : getUser.getAuthoritiesList()) {
+        roles.add(new SimpleGrantedAuthority(auth.getAuthority()));
+      }
+
+      return new UsernamePasswordAuthenticationToken(user_id, user_pw, roles);
+    } else {
+      throw new BadCredentialsException("Bad credentials");
+    }
+  }
+
+  private boolean checkSystemUser(String user_id, String user_pw) {
+    T2Users param = new T2Users();
+    param.setUserId(user_id);
+
+    return userService.checkPassword(user_pw, param);
+  }
+
+  private boolean checkByADUser(String user_id, String user_pw) {
+    boolean isAuthenticated = false;
+
+    if (StringUtil.isNotEmpty(user_pw)) {
+
+      // 사용자 가입여부 체크
+      if (!userService.existUserIdOrEmail(user_id)) {
+        T2Users vo = new T2Users();
+        vo.setUserId(user_id);
+        vo.setCreatedDateCurrentTime();
+        vo.setCreator(user_id);
+        vo.setModifier(user_id);
+
+        log.warn("Try to find Ldap user information : " + vo.getUserId());
+        String[] info = userService.checkUserInfo(vo);
+        String userName = info[0];
+        String userEmail = info[1];
+
+        if (StringUtil.isEmptyTrimmed(userEmail) || StringUtil.isEmptyTrimmed(userName)) {
+          log.warn("Cannot find Ldap user information : " + vo.getUserId());
+          return false;
         } else {
-        	loginSuccess = checkSystemUser(user_id, user_pw);
+          log.warn("Found -- Ldap user information : " + vo.getUserId());
         }
-        
-        if(loginSuccess) {
-            List<GrantedAuthority> roles = new ArrayList<GrantedAuthority>();
-            T2Users user = new T2Users();
-            user.setUserId(user_id);
-            T2Users getUser = userService.getUserAndAuthorities(user);
-            
-            for(T2Authorities auth : getUser.getAuthoritiesList()) {
-            	roles.add(new SimpleGrantedAuthority(auth.getAuthority()));
-            }
-            
-            return new UsernamePasswordAuthenticationToken(user_id, user_pw, roles);          
-        } else {
-            throw new BadCredentialsException("Bad credentials");
+
+        vo.setEmail(userEmail);
+        vo.setUserName(userName);
+        vo.setDivision(CoConstDef.CD_USER_DIVISION_EMPTY);
+        userService.addNewUsers(vo);
+      }
+
+      String principal = String.format(
+              "%s=%s,%s",
+              CoCodeManager.getCodeExpString(CoConstDef.CD_LOGIN_SETTING, CoConstDef.CD_LDAP_UID),
+              user_id,
+              CoCodeManager.getCodeExpString(CoConstDef.CD_LOGIN_SETTING, CoConstDef.CD_LDAP_BASE_DN)
+              );
+

```

Comment:
Make principle. (e.g uid=user,dc=fosslight,dc=org). LDAP_UID is set by client. You can set cn or uid or otherwise.